mcstatus/pinger.py

```diff
@@ -253,10 +253,13 @@ def _parse_description(raw_description: Union[dict, list, str]) -> str:
         for entry in entries:
             for style_key, style_val in STYLE_MAP.items():
                 if entry.get(style_key):
-                    if isinstance(style_val, dict):
-                        style_val = style_val[entry[style_key]]
+                    try:
+                        if isinstance(style_val, dict):
+                            style_val = style_val[entry[style_key]]
 
-                    description += f"§{style_val}"
+                        description += f"§{style_val}"
+                    except KeyError:
+                        pass  # ignoring these key errors strips out html color codes
```

Comment:
I'd actually convert the while to a simple if, there won't ever be nested dicts there and if there were, this logic wouldn't make sense anyway as it would only support one and then go to keyerror

Reply:
This parsing function outputs a single string. I'm not sure how to embed HTML color codes aside from outputting HTML. Perhaps, mcstatus should be formatting an HTML output? Another idea is to make use of 24-bit color escape codes to have a beautiful terminal output. https://github.com/termstandard/colors
Obviously, not all terminals can handle all the styles, but I think most can support true color at this point.

Reply:
It would be really cool to actually see the output colored, that's a great idea!

Though there still should be a way to store the description before it's printed, I don't really like the idea of just storing the description as a string with the ANSI sequences in it, even though it is a well established standard, storing something that's actually readable makes more sense imo, it also makes it a lot more usable if someone wanted to use this description somewhere where ANSI sequences don't work (such as implementing it into a website).

So instead, I'd suggest a simple list of tuples with rgb color and text, like: `[((255, 255, 255), "This text is white"), ((0, 0, 0), "This text is black")]`, however that doesn't cover everything (specifically the strikethrough, underline, etc). To cover that, we could simply additionally store some bool values (though I'd suggest a namedtuple if we'd do that since it makes it obvious which bool val is controlling which style).

We could just do something like: 
```py
from typing import NamedTuple

class StyleTuple(NamedTuple):
    rgb: tuple[int, int, int]
    underline: bool
    ...

def get_ansi_rgb(red: int, green: int, blue: int, text: str) -> str:
    return f"\033[38;2;{red};{green};{blue}m{text} \033[38;2;255;255;255m"

def parse_ansi(style_tuple: StyleTuple, text: str) -> str:
    """Returns a string with ANSI sequences as colors, based on the style tuple."""
    txt = get_ansi_rgb(style_tuple.rgb, text)
    ... # Parse the rest of the style tuple into ANSI seqs. (strikethrough, ...)
    return txt

class PingResponse:
    @staticmethod
    def _parse_description(raw_description: Union[dict, list, str]) -> list[StyleTuple, str]:
        ...  # Logic to get the list of tuples with the style and text instead of just a string
        # The output of this should later be set as self.description
    
    @property
    def ansi_description(self) -> str:
        # Note: we could also just compute this once in init (either directly or just by making this a 
        # staticmethod and calling it from init) to avoid unnecessarily doing this every time it's called, 
        # though it shouldn't take long and by having this as a property it's safe to mutate the description
        output_parts = [parse_ansi(style_tuple, text) for style_tuple, text in self.description]
        return "".join(output_parts)
```

Reply:
Is what you are proposing a slightly different version of the input? The input is already a dictionary that groups the styles in this manner https://minecraft.fandom.com/wiki/Raw_JSON_text_format

Reply:
Oh nevermind, I get it now. We would want to convert the ANSI escapes into the same format for consistency, because of the union type on the raw description.

Reply:
This could wait for a later PR. For now ignoring the HTML codes is better than crashing, right?

Reply:
> This could wait for a later PR. For now ignoring the HTML codes is better than crashing, right?

Yeah, probably, you should open an issue about this explicitly if it won't be fixed here though. 

But in that case my initial comment that we should probably have a regex and only ignore the HTML color codes, not anything else stands. 

But since the colored output with ANSI sequences will get postponed, I'm not exactly sure how to embed the HTML colors to that description either, and I wouldn't mind if it was just ignored completely for now I suppose. Though it may be worth to consider even just doing something as simple as `<#XXXXXX>text`. It's not exactly a standardized format, but at least it shows that there is some color there, but again, I wouldn't mind that much if that's not added.

Reply:
#204 created